spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cell.scala

```diff
@@ -117,47 +121,66 @@ abstract class Cell[T : ClassTag](
         require(hidden.isInstanceOf[Table],
           "Cell: hidden should be a Table")
         var i = 1
-        if (null == imageSize) {
-          while (i <= hidden.toTable.length()) {
-            hidden.toTable[Tensor[T]](i).resize(batchSize, hiddensShape(i - 1))
-            i += 1
-          }
-        } else {
-          val sizes = new Array[Int](imageSize.length + 1)
-          sizes(0) = batchSize
-          Array.copy(imageSize, 0, sizes, 1, imageSize.size)
-          while (i <= hidden.toTable.length()) {
-            sizes(1) = hiddensShape(i - 1)
-            hidden.toTable[Tensor[T]](i).resize(sizes)
-            i += 1
-          }
+        val sizes = new Array[Int](stepShape.length + 1)
+        sizes(0) = batchSize
+        Array.copy(stepShape, 0, sizes, 1, stepShape.size)
+        while (i <= hidden.toTable.length()) {
+          sizes(1) = hiddensShape(i - 1)
+          hidden.toTable[Tensor[T]](i).resize(sizes)
+          i += 1
         }
         hidden
       }
     }
   }
 
   override def updateOutput(input: Table): Table = {
-    output = cell.forward(input).toTable
+    if (combinePreTopology) {
+      val inputTensor = input.toTable[Tensor[T]](Recurrent.inputDim)
+      input(Recurrent.inputDim) = preTopology.updateOutput(inputTensor)
```

Comment:
what if preTopology is null?

Reply:
combinePreTopology is set at RecurrentDecoder, if preTopology is null, combinePreTopology is using default value(false)

Reply:
So add assertion here?